lib/heap_sort.rb

```diff
@@ -1,8 +1,13 @@
 
-
 # This method uses a heap to sort an array.
 # Time Complexity:  ?
 # Space Complexity: ?
-def heap_sort(list)
-  raise NotImplementedError, "Method not implemented yet..."
-end
\ No newline at end of file
+def heapsort(list)
```

Comment:
⚠️  , Space & time complexity?

Also you add things into a heap, but in the internal structure the items are **not** sorted.  They're maintained in heap order with the parent smaller than it's children.  You have to REMOVE elements from the heap 1-by-1 to be guaranteed to get it into order.